### PR TITLE
[Service Connector] `az aks connection`: AKS support for Service Connector

### PR DESCRIPTION
**Pull request by wchigit**
**Related command**
az aks connection create/update/list/list-configuration/validate/delete

**Description**
Support AKS as source resource for Service Connector

**Testing Guide**
Create using workload identity as authentication type
az aks connection create {target-type} --connection {connection-name} --source-id {source-aks-id} --target-id {target-id} --workload-identity client-id={client-id} subs-id={subscription-id} --client-type python
az aks connection list --source-id {source-aks-id}
az aks connection list-configuration --id {connection-id}
az aks connection validate --id {connection-id}
az aks connection delete --id {connection-id}

Create using secret:
az aks connection create {target-type} --connection {connection-name} --source-id {source-aks-id} --target-id {target-id} --secret --client-type python

**History Notes**

[Service Connector] `az aks connection`: AKS support for Service Connector

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py [hunk 1]
```diff
@@ -712,6 +711,14 @@ class CLIENT_TYPE(Enum):
             'action': AddUserAssignedIdentityAuthInfo
         }
     },
+    # used only as argument, same action as user identity
+    AUTH_TYPE.WorkloadIdentity: {
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py:715]: mark aks command as preview?

File: src/azure-cli/azure/cli/command_modules/serviceconnector/action.py [hunk 2]
```diff
@@ -114,7 +119,7 @@ def get_action(self, values, option_string, command_name):  # pylint: disable=no
                 d['client_id'] = v[0]
             elif kl == 'subs-id':
                 d['subscription_id'] = v[0]
-            elif is_mysql_target(command_name) and kl == 'mysql-identity-id':
+            elif is_mysql_target(command_name) and not is_k8s_source(command_name) and kl == 'mysql-identity-id':
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/action.py:122]: will passwordless cmds fail? I remember there's  some change need in passwordless extension before aks +passwordless is fully tested. could you pls sync with Xiaofan to make sure passwordless cmds fail with not supported error before ready? then aks +passwordless support.

**Review by xfz11 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py [hunk 3]
```diff
@@ -270,6 +271,15 @@ def get_source_display_name(sourcename):
...
                 subs-id        : Required. Subscription id of the user assigned identity.
         ''' if AUTH_TYPE.UserIdentity in auth_types else ''
+        workload_identity_param = '''
+            - name: --workload-identity
+              short-summary: The Microsoft Entra Workload ID with AKS.
+              long-summary: |
+                Usage: --workload-identity client-id=XX subs-id=XX
+
+                client-id      : Required. Client id of the workload identity, i.e., a user assigned identity.
+                subs-id        : Required. Subscription id of the workload identity, i.e., a user assigned identity.
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_help.py:281]: to make it simple, let's use a `--workload-identity <User-MI-Resource-ID>`

then, let's call `run_cli_cmd` to get the `client_id` and `subs_id` of the identity and pass it to client

param description: `An user-assigned managed identity used to create workload identity federation. (Please DO NOT use AKS control plane identity and kubelet identity which is not supported by federated identity credential.) `

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py [hunk 4]
```diff
@@ -828,23 +835,23 @@ class CLIENT_TYPE(Enum):
...
-        RESOURCE.StorageTable: [AUTH_TYPE.SecretAuto, AUTH_TYPE.ServicePrincipalSecret],
+        RESOURCE.StorageTable: [AUTH_TYPE.WorkloadIdentity, AUTH_TYPE.SecretAuto, AUTH_TYPE.ServicePrincipalSecret],
 
-        RESOURCE.KeyVault: [AUTH_TYPE.ServicePrincipalSecret],
-        RESOURCE.AppConfig: [AUTH_TYPE.SecretAuto, AUTH_TYPE.ServicePrincipalSecret],
-        RESOURCE.EventHub: [AUTH_TYPE.SecretAuto, AUTH_TYPE.ServicePrincipalSecret],
-        RESOURCE.ServiceBus: [AUTH_TYPE.SecretAuto, AUTH_TYPE.ServicePrincipalSecret],
-        RESOURCE.SignalR: [AUTH_TYPE.SecretAuto, AUTH_TYPE.ServicePrincipalSecret],
-        RESOURCE.WebPubSub: [AUTH_TYPE.SecretAuto, AUTH_TYPE.ServicePrincipalSecret],
+        RESOURCE.KeyVault: [AUTH_TYPE.WorkloadIdentity, AUTH_TYPE.ServicePrincipalSecret],
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py:849]: we also need add a `--secret-store-csi-driver` auth type for keyvault, which takes no arguments.

it enables csi-driver for keyvault

**Comment by yonzhan** [comment]
Please fix CI issues

**Review by houk-ms — APPROVED** [review]
good work

**Comment by wchigit** [comment]
@jsntcy Could you please help review this PR?

**Comment by jsntcy** [comment]
@kairu-ms please help review

**Comment by wchigit** [comment]
Need to disable secret store.

**Review by zhoxing-ms — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**